packages/react-native/ReactAndroid/src/main/java/com/facebook/react/HeadlessJsTaskService.java

```diff
@@ -166,4 +161,55 @@ public void onHeadlessJsTaskFinish(int taskId) {
   protected ReactNativeHost getReactNativeHost() {
     return ((ReactApplication) getApplication()).getReactNativeHost();
   }
+
+  /**
+   * Get the {@link ReactHost} used by this app. By default, assumes {@link #getApplication()}
+   * is an instance of {@link ReactApplication} and calls {@link
+   * ReactApplication#getReactHost()}.
+   * This method assumes it is called in new architecture and returns null if not.
+   */
+  protected ReactHost getReactHost() {
+    return ((ReactApplication) getApplication()).getReactHost();
+  }
+
+  protected ReactContext getReactContext() {
+    if (DefaultNewArchitectureEntryPoint.getBridgelessEnabled()) {
```

Comment:
Can we use this instead : `BuildConfig.IS_NEW_ARCHITECTURE_ENABLED` ?

Reply:
Nope you won't be able to use `BuildConfig.IS_NEW_ARCHITECTURE_ENABLED` as that is available only inside app/library code, and not inside the framework

Reply:
But the problem with this is the circular dependency it will lead to :
`react -> defaults -> react`